cmake/toolchain.linux-x64-asan.cmake

```diff
@@ -0,0 +1,55 @@
+# Toolchain for compiling with ASAN enabled on a Linux-x86-64 host.
+# This is done as a "crosscompile" because we must use our LLVM version
+# of clang for *all* compilation (rather than using it just for bitcode
+# and letting the host compiler, eg gcc, handle everything else); ASAN
+# essentially requires everything to be compiled with matching versions
+# of the same compiler.
+#
+# Note: requires LLVM to be built with -DLLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi;libunwind"
+#
+# Note: only tested with LLVM/Clang 16 as of this comment. Earlier versions
+# may likely work but are not tested.
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR i686)
+
+set(LLVM_INSTALL_DIR ${LLVM_DIR}/../../../)
```

Comment:
Should `FATAL_ERROR` if `LLVM_DIR` is not set

Reply:
Interestingly, that will make things fail: apparently we execute this code again during the 'Detecting C compiler ABI info' phase, but without the value set, so we just have to hope for the best. (If it's not set, the compilation will fail anyway, of course)